components/scream/src/physics/shoc/tests/shoc_run_and_cmp.cpp

```diff
@@ -15,44 +15,78 @@ using namespace scream;
 using namespace scream::util;
 using namespace scream::shoc;
 
+  /* shoc_run_and_cmp can be run in 2 modes. First, generate_baseline 
+   * runs the baseline (aka reference, probably git master) version of 
+   * the code and saves its output as a raw binary file. Then run_and_cmp
+   * runs the new/experimental version of the code and compares it against
+   * the baseline data you've saved to file. Both baseline and cmp modes
+   * start from an initial condition in ../shoc_ic_cases.cpp. Each call to 
+   * shoc_main loops through nadv=15 steps with dt=5 min. On top of this, 
+   * shoc_main is called iteratively num_iters=10 steps, performing checks 
+   * and potentiallywriting output each time. This means that shoc_run_and_cmp 
+   * is really a single 150-step shoc run.
+   */
+  
+
+/* Given a column of data for variable "label" from the reference run 
+ * (probably master) and from your new exploratory run, loop over all 
+ * heights and confirm whether or not the relative difference between  
+ * runs is within tolerance "tol". If not, print debug info. Here, "a"
+ * is the value from the reference run and "b" is from the new run.
+ */
 template <typename Scalar>
```

Comment:
@jeff-cohere - is the positioning and formatting of my function comment correct now? Thanks!

Reply:
Yep, looks great! Thanks for doing it in the P3 version, too.